### PR TITLE
Avoid a cyclic reference while printing EngineExchangeTransitionConfigurationParameter

### LINKED ISSUE #4354: ERROR | ExecutionEngineJsonRpcMethod | failed to exec consensus method engine_exchangeTransitionConfigurationV1
When turning on TRACE the ERROR " failed to exec consensus method engine_exchangeTransitionConfigurationV1" It is unclear whether this will result into merge issues to me. See ther error below:

sep 07 14:27:23 ssa-Blade besu[207075]: 2022-09-07 14:27:23.028+02:00 | vert.x-worker-thread-0 | DEBUG | ExecutionEngineJsonRpcMethod | execution engine JSON-RPC request engine_exchangeTransitionConfigurationV1 [{terminalTotalDifficulty=0xc70d808a128d7380000, terminalBlockHash=0x0000000000000000000000000000000000000000000000000000000000000000, terminalBlockNumber=0x0}]
sep 07 14:27:23 ssa-Blade besu[207075]: 2022-09-07 14:27:23.028+02:00 | vert.x-eventloop-thread-17 | ERROR | ExecutionEngineJsonRpcMethod | failed to exec consensus method engine_exchangeTransitionConfigurationV1
sep 07 14:27:23 ssa-Blade besu[207075]: io.vertx.core.json.EncodeException: Failed to encode as JSON: Direct self-reference leading to cycle (through reference chain: org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.EngineExchangeTransitionConfigurationParameter["terminalTotalDifficulty"]->org.hyperledger.besu.ethereum.core.Difficulty["asBytes32"])
sep 07 14:27:23 ssa-Blade besu[207075]:         at io.vertx.core.json.jackson.DatabindCodec.toString(DatabindCodec.java:163)
sep 07 14:27:23 ssa-Blade besu[207075]:         at io.vertx.core.json.Json.encodePrettily(Json.java:71)
sep 07 14:27:23 ssa-Blade besu[207075]:         at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine.EngineExchangeTransitionConfiguration.lambda$syncResponse$1(EngineExchangeTransitionConfiguration.java:84)
sep 07 14:27:23 ssa-Blade besu[207075]:         at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
sep 07 14:27:23 ssa-Blade besu[207075]:         at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
sep 07 14:27:23 ssa-Blade besu[207075]:         at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
sep 07 14:27:23 ssa-Blade besu[207075]:         at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
sep 07 14:27:23 ssa-Blade besu[207075]:         at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
sep 07 14:27:23 ssa-Blade besu[207075]:         at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
sep 07 14:27:23 ssa-Blade besu[207075]:         at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
sep 07 14:27:23 ssa-Blade besu[207075]:         at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
sep 07 14:27:23 ssa-Blade besu[207075]:         at org.hyperledger.besu.util.Slf4jLambdaHelper.traceLambda(Slf4jLambdaHelper.java:54)
sep 07 14:27:23 ssa-Blade besu[207075]:         at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine.EngineExchangeTransitionConfiguration.syncResponse(EngineExchangeTransitionConfiguration.java:81)
sep 07 14:27:23 ssa-Blade besu[207075]:         at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.ExecutionEngineJsonRpcMethod.lambda$response$0(ExecutionEngineJsonRpcMethod.java:67)
sep 07 14:27:23 ssa-Blade besu[207075]:         at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
sep 07 14:27:23 ssa-Blade besu[207075]:         at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
sep 07 14:27:23 ssa-Blade besu[207075]:         at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
sep 07 14:27:23 ssa-Blade besu[207075]:         at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
sep 07 14:27:23 ssa-Blade besu[207075]:         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
sep 07 14:27:23 ssa-Blade besu[207075]:         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
sep 07 14:27:23 ssa-Blade besu[207075]:         at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
sep 07 14:27:23 ssa-Blade besu[207075]:         at java.base/java.lang.Thread.run(Thread.java:833)

### PR DESCRIPTION
**Pull request by daniellehrner**
Signed-off-by: Daniel Lehrner <daniel.lehrner@consensys.net>

## PR description

EngineExchangeTransitionConfigurationParameter is printed if logging is set to `TRACE`. In the current version it throws an exception while trying to print it, because of a circular reference. This PR avoids this circular reference and EngineExchangeTransitionConfigurationParameter is printed as expected.

## Fixed Issue(s)
fixes #4354

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Review by macfarla — APPROVED** [review]
LGTM but spotless sad

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/EngineExchangeTransitionConfigurationParameter.java [hunk 1]
```diff
@@ -46,4 +46,13 @@ public Hash getTerminalBlockHash() {
...
   }
+
+  @Override
+  public String toString() {
+    return "EngineExchangeTransitionConfigurationParameter{" +
+            "terminalTotalDifficulty=" + terminalTotalDifficulty.toShortHexString() +
+            ", terminalBlockHash=" + terminalBlockHash.toShortHexString() +
+            ", terminalBlockNumber=" + terminalBlockNumber +
+            '}';
+  }
```
Comment (fab-10) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/EngineExchangeTransitionConfigurationParameter.java:57]: Switching to Jackson this is not needed, btw `toString` is free form and could not return a valid json if modified in the future, better use something else.

**Review by fab-10 — CHANGES_REQUESTED** [review]
I suggest to use Jackson, so manual serialization can be avoided, and is more future proof, and
please update the CHANGELOG

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/EngineExchangeTransitionConfigurationParameter.java (hunk 1 above)
Comment (daniellehrner) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/EngineExchangeTransitionConfigurationParameter.java]: I've updated the PR to use Jackson

**Review by fab-10 — APPROVED** [review]
LGTM, just add a CHANGELOG entry

**Pull request merged**